### DISCUSSION TITLE
Connecting a development instance of mod-ui with a production version

### DISCUSSION THREAD

**Discussion by fritzvd** [mod-audio/mod-ui#148] [Q&A] [answered]
Is it possible to connect a mod dwarf or mod desktop instance with a development instance of mod-ui?

I'm trying to get started working on the UI a bit (out of general interest, but also was hoping to figure out dropping in fx in existing chains)

**Comment by falkTX** [comment]
that is what https://github.com/moddevices/mod-ui/blob/master/mod-deploy.sh is for

**Reply by fritzvd** [reply]
Right, so if I understand correctly, I can use mod-deploy, to deploy a development version to a "real" device and see if it works?

**Reply by falkTX** [reply, marked as answer]
exactly. just beware to use a hotfix branch that matches the version on the device, for the best compat.

**Reply by fritzvd** [reply]
And just to be sure. The ui runs off of the device right?

**Reply by falkTX** [reply]
yes, webserver is always on device side. needs to be in order to show the correct plugin resources

**Reply by fritzvd** [reply]
Ok, so if I run development on a local machine I have to run mod-ui and mod-host, right? But can't connect a mod-desktop instance to a development version of mod-ui

p.s. thanks for responding 💜

**Reply by fritzvd** [reply]
Let me clarify why I'm asking. whenever I try to run ui and host in development, I get an empty pedalboard and no plugins. So I thought I might be able to connect my existing mod-desktop instance. Anywsy, perhaps you know why or what is wrong? Maybe something I missed?

<img width="1624" alt="Screenshot 2024-05-07 at 15 11 26" src="https://github.com/moddevices/mod-ui/assets/160328/50f345e5-fc0d-44c8-86b5-cb7738d4af0e">

**Reply by fritzvd** [reply]
@falkTX any pointers on how to get started with getting the plugins to load in a local version? I can't really find documentation about this.
